pybricks/common/pb_type_speaker.c

```diff
@@ -22,34 +23,90 @@
 #include <pybricks/util_mp/pb_obj_helper.h>
 #include <pybricks/util_pb/pb_error.h>
 
+#define MP_PI MICROPY_FLOAT_CONST(3.14159265358979323846)
+
 typedef struct {
     mp_obj_base_t base;
     bool intialized;
 } pb_type_Speaker_obj_t;
 
 STATIC pb_type_Speaker_obj_t pb_type_Speaker_singleton;
 
+STATIC uint8_t volume = 100; // volume in 0..100 range
+
+// The number to multiply the sample amplitude by, to amplify the signal by the defined volume.
+// The original sample amplitude must be in the -1..1 range.
+STATIC uint16_t sample_amplifier = INT16_MAX;
```

Comment:
We should probably make these members of `pb_type_Speaker_obj_t` instead of global variables.

Reply:
How would the `STATIC` methods have access to instance variables? In the Java world, this would not work for static methods.

Reply:
For the C functions that are MicroPython methods, through the `self` variable. For other functions, it should be passed as a parameter.

Reply:
Done.

Reply:
I imagine that we will eventually have OS sounds and that the OS will take care of resetting the volume on each program run. 

> Wondering what would happen if a Python program creates two Speaker instances though.

For edge cases like this, I usually add a `// REVISIT: ...` comment that explains what I was thinking at the time. Then we can look at it again when we have better infrastructure to handle such things. Currently our cleanup/reset when a user program ends is a bit of a hack.

Reply:
OK. Will add `REVISIT` comment let the volume be reset upon every instance creation.

Reply:
Done.